src/main/java/net/patchworkmc/patcher/patch/redirect/NewToFactoryRedirector.java

```diff
@@ -0,0 +1,69 @@
+package net.patchworkmc.patcher.patch.redirect;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.tree.TypeInsnNode;
+
+import net.patchworkmc.patcher.patch.util.ClassRedirection;
+import net.patchworkmc.patcher.transformer.NodeTransformer;
+
+public class NewToFactoryRedirector extends NodeTransformer {
```

Comment:
Having a comment like in VirtualToStaticRedirector here would be helpful, it took me a bit to figure out what this was actually doing.